rust-src/concordium_base/src/bulletproofs/range_proof.rs

```diff
@@ -185,6 +189,15 @@ pub fn prove<C: Curve, T: Rng>(
         transcript.append_message(b"Vj", &V_j.0);
         V_vec.push(V_j);
     }
+
+    if let ProofVersion::Version2 = version {
```

Comment:
If we ever add proof version 3 this might be difficult to notice.

I'd recommend deriving `PartialOrd` and `Ord` and then using `>= Version2` as the condition.